string topicname
uint32 messagecount